src/builder-post-process.c

```diff
@@ -377,9 +377,21 @@ builder_post_process_debuginfo (GFile          *app_dir,
         }
       else if (g_str_has_prefix (rel_path_dir, "usr/"))
         {
-          debug_dir = g_build_filename (app_dir_path, "usr/lib/debug", rel_path_dir, NULL);
-          real_debug_dir = g_build_filename ("/usr/lib/debug", rel_path_dir, NULL);
-          source_dir_path = g_build_filename (app_dir_path, "usr/lib/debug/source", NULL);
+          BuilderOptions * options = builder_context_get_options (context);
+          const char * libdir = builder_options_get_libdir (options, context);
+
+          if (libdir)
+            {
+              debug_dir = g_build_filename (app_dir_path, libdir, "debug", rel_path_dir, NULL);
+              real_debug_dir = g_build_filename (libdir, "debug", rel_path_dir, NULL);
+              source_dir_path = g_build_filename (app_dir_path, libdir, "debug/source", NULL);
```

Comment:
Same here. There only ever need to be one prefix for "where the debuginfo is stored", which is /usr/lib/debug. And then multiarch works fine because the path is stored below that.